libp2p/pubsub/mcache.py

```diff
@@ -96,7 +96,8 @@ def shift(self) -> None:
         last_entries: List[CacheEntry] = self.history[len(self.history) - 1]
 
         for entry in last_entries:
-            del self.msgs[entry.mid]
+            if entry.mid in self.msgs:
```

Comment:
It's a minimal improvement but you can also do `self.msgs.pop(encry.mid, None)`.  Slightly more robust as it's also threadsafe though admittedly that probably doesn't matter much here.